### DISCUSSION TITLE
Running Oracle 19c with ARM Binaries in Standalone mode

### DISCUSSION THREAD

**Discussion by bm-enverus** [oracle/docker-images#2720] [General]
The DockerFile provided spins up and bootstraps a pluggable database. I have a current database using standalone in OFA that i would like to mount into the container. I copy all of my database files into a u02 directory, but oracle only detects the newly created pluggable database. What is the process to have oracle detect my standalone database?

**Comment by oraclesean** [comment]
By standalone, do you mean a non-CDB database? If so, there are several things to consider. I assume the database you're trying to start is also a 19c database. If so, this is a high-level set of steps.

I recommend creating the container with volumes for the default `/opt/oracle/oradata` data directory and the new, '/u02` volume you're using for the new datafiles. Volume associations can only be created when the container's first started, via `docker run`.

Next, tell the container about the new database via its metadata and configuration. This includes some files your original database stores in `ORACLE_HOME/dbs` and `ORACLE_HOME/network/admin` (assuming it doesn't use a Read-Only Home). You'll need to copy the password file, spfile or pfile, `tnsnames.ora` and `sqlnet.ora` (and any wallets, etc) to a dedicated directory under the container volume, `/opt/oracle/oradata/dbconfig/$ORACLE_SID`.

Once those files are there, you must create soft-links in the container's `ORACLE_HOME` that reference their location in the configuration directory. You can look through the repository scripts for examples of these commands.

Then, add an entry for the new database in `/etc/oratab`. 

Now you can call `.oraenv` to set the environment, passing the new database's `ORACLE_SID`, open a SQL*Plus session, and start the database. Since the container won't be aware of the new database via its entrypoints, you'll need to start this database each time the container starts (or modify the `startDB.sh` scripts in the container itself).

Another option is to alter the repository files by modifying the dbca template to create a non-CDB database, change the `startDB.sh` script to skip the PDB startup, and (optionally) update the data volume from `/opt/oracle/oradata` to `/u02`. Use this to create a new image, then run a container, specifying the `ORACLE_SID` of your database and mount the `/u02` path to a new, empty directory on the local machine. Run the container and let it create all the necessary files, then stop it and replace what it creates with the files from the database you're transferring.

You can achieve similar results via RMAN, a Data Pump export/import, or even a PDB plug-in operation.
